pleskdistup/actions/systemd.py

```diff
@@ -147,6 +147,37 @@ def estimate_revert_time(self) -> int:
         return 10
 
 
+class HandlePleskFirewallService(action.ActiveAction):
+    def __init__(self) -> None:
+        self.name = "handle plesk-firewall service"
+        self.plesk_firewall_service = "plesk-firewall.service"
+
+    def _is_required(self) -> bool:
+        return systemd.is_service_startable(self.plesk_firewall_service) and systemd.is_service_active(self.plesk_firewall_service)
+
+    def _prepare_action(self) -> action.ActionResult:
+        # The firewalld service conflicts with the plesk-firewall service, so we need to ensure
+        # that it does not start on the target system
+        if systemd.is_service_exists("firewalld.service"):
+            util.logged_check_call(["/usr/bin/systemctl", "stop", "firewalld.service"])
+            util.logged_check_call(["/usr/bin/systemctl", "disable", "firewalld.service"])
+
+        util.logged_check_call(["/usr/bin/systemctl", "stop", "plesk-firewall.service"])
+        util.logged_check_call(["/usr/bin/systemctl", "disable", "plesk-firewall.service"])
+        return action.ActionResult()
+
+    def _post_action(self) -> action.ActionResult:
+        util.logged_check_call(["/usr/bin/systemctl", "enable", "plesk-firewall.service"])
+        # Don't do startup because the services will be started up after reboot at the end of the script anyway.
+        return action.ActionResult()
+
+    def _revert_action(self) -> action.ActionResult:
```

Comment:
Maybe remember initial `firewalld` state and restore it on revert too?

Reply:
It's not obvious, because firewalld and plesk-firewall are not meant to operate at the same time. If both are enabled, something is likely wrong.
On the other hand centos2alma was not designed to fix such issues by reverting operations. Therefore, it might be better to return the server to its original state, even if that state is incorrect :thinking:

Reply:
Ah, wait a moment. Plesk-firewall already performs the same function. Therefore, this approach may not be the best. I will need to revise the change